pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -322,6 +326,34 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{RequeueAfter: 5 * time.Second}, nil
 	}
 
+	// Check if enough routes exist
+	routesExist, err := r.routesExist()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if zync deployments are ready
+	zyncReady, err := r.isZyncReady()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// If zync is enabled and zync is ready but the routes are missing then create the ResyncRoutes Job
+	if system.Options.ZyncEnabled && zyncReady && !routesExist {
```

Comment:
I know we are making the call against system..but this feels to me like a zync reconciler task rather than system task..

Reply:
Then, I guess we need to check if system is ready right? Also, does the system-reconciler completes when system is up and running OR when system deployment reports completed? 
I guess what I'm asking is - is there a chance that we are going to run the re-sync job against non-working system?

Reply:
No there's no risk of running the re-sync Job against a non-working system because [this block](https://github.com/3scale/3scale-operator/blob/master/pkg/3scale/amp/operator/system_reconciler.go#L324-L328) requeues the reconciler until `system-app` is fully ready. I initially had the Job in the zync reconciler but I thought it was better suited for the system reconciler because the command is run in a porta (`system-sidekiq`) image.

Reply:
thanks, IMO this should be a part of "Enable Zync" function etc, but, no strong opinion, I leave it up to you.